src/lib/y2partitioner/widgets/disk_table.rb

```diff
@@ -22,13 +22,15 @@ def initialize(disks, pager)
 
       # table items. See CWM::Table#items
       def items
+        probed_graph = Y2Storage::StorageManager.instance.y2storage_probed
```

Comment:
That is what I had in mind, when talking about work in a copy. Basically, have the probed graph and the staging graph, and copy the staging graph to a proposal graph.

So basically before any action that modifies the graph, copy the staging graph to the proposal graph and if finish commit the changes (copy proposal to staging or revert (copy staging to proposal)) and redraw.